src/app/pages/charging-stations/details-component/charging-stations-start-transaction-details-dialog-component.ts

```diff
@@ -0,0 +1,243 @@
+                this.spinnerCounter--;
+                if (this.spinnerCounter === 0) {
+                    this.spinnerService.hide();
+                }
+                switch (error.status) {
+                    case HTTPError.OBJECT_DOES_NOT_EXIST_ERROR:
+                        this.car.disable();
+                        break;
+                    default:
+                        Utils.handleHttpError(error, this.router, this.messageService,
+                            this.centralServerService, 'cars.car_error');
+                }
+            });
+        }
+    }
+
+    public assignUser() {
+        const dialogConfig = new MatDialogConfig();
+        dialogConfig.panelClass = 'transparent-dialog-container';
+        // Set data
+        dialogConfig.data = {
+            rowMultipleSelection: false,
+            staticFilter: {
+            },
+        };
+        // Show
+        const dialogRef = this.dialog.open(UsersDialogComponent, dialogConfig);
+        // Register to the answer
+        dialogRef.afterClosed().subscribe((result) => {
+            this.user.setValue(Utils.buildUserFullName(result[0].objectRef));
+            this.userID.setValue(result[0].key);
+            this.tag.setValue('');
+            this.tagID.setValue('');
+            this.car.setValue('');
+            this.carID.setValue('');
+            this.spinnerCounter = 2;
+            this.loadDefaultCar();
+            this.loadDefaultTag();
+        });
+    }
+
+    public assignTag() {
+        const dialogConfig = new MatDialogConfig();
+        dialogConfig.panelClass = 'transparent-dialog-container';
+        // Set data
+        dialogConfig.data = {
+            rowMultipleSelection: false,
+            staticFilter: {
+                UserID: this.userID.value,
+                Active: true
+            },
+        };
+        // Show
+        const dialogRef = this.dialog.open(TagsDialogComponent, dialogConfig);
+        // Register to the answer
+        dialogRef.afterClosed().subscribe((result) => {
+            this.tag.setValue(result[0].key);
+            this.tagID.setValue(result[0].key);
+        });
+    }
+
+    public assignCar() {
+        const dialogConfig = new MatDialogConfig();
+        dialogConfig.panelClass = 'transparent-dialog-container';
+        // Set data
+        dialogConfig.data = {
+            rowMultipleSelection: false,
+            staticFilter: {
+                UserID: this.userID.value
+            },
+        };
+        // Show
+        const dialogRef = this.dialog.open(CarsDialogComponent, dialogConfig);
+        // Register to the answer
+        dialogRef.afterClosed().subscribe((result) => {
+            this.car.setValue(Utils.buildCarName(result[0].objectRef, false));
+            this.carID.setValue(result[0].key);
+        });
+    }
+
+    public startTransaction() {
+        const startTransaction: StartTransaction = {
+            userFullName: this.user.value,
```

Comment:
This should be userID right?

Reply:
We only need the userFullName to display in the confirmation pop up. if we pass the userID we will need to call the server to get the Object.

Reply:
But why is there a carID and tagID below in that case?

Reply:
We pass the carID, TagID and userFullName to the method startTransactionForUser in the table-charging-stations-start-transaction-action. 
- CarID and TagID go to the server to start transaction 
- userFullName to display in the confirmation Modal

Reply:
Ok got it, but when the admin chooses another user, you should also pass the user ID right?

Reply:
No, only the TagID go to the server in all cases.

Reply:
We always retrieve the user from tag ID.